src/cryoemservices/wrappers/clem_process_raw_tiffs.py

```diff
@@ -0,0 +1,402 @@
+        xml_file = series_path.parent / "Metadata" / (series_name_short + ".xlif")
+        if xml_file.exists():
+            logger.info(f"Metadata file found at {xml_file}")
+        else:
+            logger.error(f"No metadata file found at {xml_file}")
+            return None
+    else:
+        xml_file = metadata_file
+
+    # Process TIFF files and collect results
+    results = process_tiff_files(tiff_list, xml_file, save_dir)
+
+    return results
+
+
+class TIFFToStackParameters(BaseModel):
+    """
+    Pydantic model for validating the received message for the TIFF file conversion
+    workflow.
+
+    The keys under the "job_parameters" key in the Zocalo wrapper recipe must match
+    the attributes present in this model. Attributes in the model with a default
+    value don't have to be provided in the wrapper recipe.
+    """
+
+    tiff_list: Union[list[Path], str, Literal["null"]]
+    root_folder: str
+    metadata: Path
+    tiff_file: Union[Path, Literal["null"]] = (
+        "null"  # Alternative input when testing wrapper individually
+    )
+
+
+class TIFFToStackWrapper(BaseWrapper):
+    def run(self) -> bool:
+        assert hasattr(self, "recwrap"), "No RecipeWrapper object found"
+        params_dict = self.recwrap.recipe_step["job_parameters"]
+        try:
+            params = TIFFToStackParameters(**params_dict)
+        except (ValidationError, TypeError) as e:
+            logger.error(
+                f"TIFFToStackParameters validation failed for parameters: {params_dict} with exception: {e}"
+            )
+            return False
+
+        # Set up parameters
+        tiff_file = params.tiff_file
+        root_folder = params.root_folder
+
+        # If no file list is provided, use a singular TIFF file to get the necessary files
+        if isinstance(params.tiff_list, str):
+            # If "null" is set, use 'tiff_file' parameter to construct list
+            if params.tiff_list == "null":
+                if tiff_file == "null":
+                    logger.error(
+                        "'tiff_file' cannot be 'null' if 'tiff_list' is already 'null'"
+                    )
+                    return False
+                elif isinstance(tiff_file, Path):
+                    tiff_list = [
+                        f.resolve()
+                        for f in tiff_file.parent.glob("./*")
+                        if f.suffix in {".tif", ".tiff"}
+                        # Handle cases where series start with the same position number,
+                        # but deviate afterwards
+                        and f.stem.startswith(tiff_file.stem.split("--")[0] + "--")
+                    ]
+                else:
+                    logger.error("Error parsing 'tiff_file' parameter")
+                    return False
+            # Check if it's a stringified list
+            elif (
+                params.tiff_list.startswith("['") and params.tiff_list.endswith("']")
```

Comment:
Do you need the conditions on [' and ["? Is it not just ok to check that the first item is [?